addons/web/static/src/oxp/todo_item.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+
+  <t t-name="TodoItem" >
+    <t t-if="state.isInEdition">
+        <input t-att-value="props.todoItem.message" t-on-change="editMessage" class="form-control"/>
+    </t>
+    <t t-else="">
+        <span t-esc="props.todoItem.message"
+          t-att-class="props.todoItem.isDone ? 'text-decoration-line-through' : ''"
+          t-on-click="() => this.toggleDone()"/>
+        <div>
+          <button t-on-click="() => this.switchInEdition()" type="button" class="btn btn-primary me-1">Edit</button>
```

Comment:
here, and elsewhere, I usually don't create lambda function if not necessary: `t-on-click="switchInEdition`. But then, it may be annoying in some cases when using it as a prop because you need `.bind`